### PR TITLE
time tracker

### PR DESCRIPTION
**Pull request by narkarum**

### CODE REVIEW

File: volume-providers/vpc/vpcclient/vpcvolume/util.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	pc, _, _, _ := runtime.Caller(1)
+
+	// Retrieve a function object this functions parent.
+	funcObj := runtime.FuncForPC(pc)
+
+	// Regex to extract just the function name (and not the module path).
+	runtimeFunc := regexp.MustCompile(`^.*\.(.*)$`)
+	name := runtimeFunc.ReplaceAllString(funcObj.Name(), "$1")
+
+	log.Println(fmt.Sprintf("%s took %s", name, elapsed))
```
Comment (prgavali) [inline volume-providers/vpc/vpcclient/vpcvolume/util.go]: Can we use the caller`s zap logger here? We need to pass `logger ` to TimeTrack() method .
Also , I'm in favour of passing `function Name ` as a parameter to this method. Finding the the function from `runtime`  dynamically is quite expensive . This method will be used by  lot of method . So it needs to be optimised

Comment (narkarum) [reply volume-providers/vpc/vpcclient/vpcvolume/util.go]: Addressed

**Comment by narkarum** [comment]
@pramodgavali Can you check now?

Addressed the comments.

**Comment by narkarum** [comment]
```
{"level":"info","timestamp":"2019-04-05T11:02:32.163+0530","caller":"provider/delete_volume.go:21","msg":"Entry DeleteVolume()","name":"ibmcloud-storage-volume-lib/main","VolumeLib":"IBMCloud-Storage-Lib","vol":{"volumeID":"0f943442-4c42-4e07-beca-cde25c64e5f5","provider":"","volumeType":"","capacity":null,"iops":null,"tier":null,"creationTime":"0001-01-01T00:00:00Z","serviceOffering":null}}
{"level":"info","timestamp":"2019-04-05T11:02:32.163+0530","caller":"provider/get_volume.go:23","msg":"Entry GetVolume","name":"ibmcloud-storage-volume-lib/main","VolumeLib":"IBMCloud-Storage-Lib"}
2019/04/05 11:02:34 GetVolume took 2.383132788s
{"level":"info","timestamp":"2019-04-05T11:02:34.547+0530","caller":"provider/get_volume.go:38","msg":"Volume details","name":"ibmcloud-storage-volume-lib/main","VolumeLib":"IBMCloud-Storage-Lib","Volume":{"href":"https://us-south.iaas.cloud.ibm.com/v1/volumes/0f943442-4c42-4e07-beca-cde25c64e5f5","id":"0f943442-4c42-4e07-beca-cde25c64e5f5","name":"simha13","capacity":10,"iops":3000,"resource_group":{"href":"https://resource-manager.bluemix.net/v1/resource_groups/efd16c8eac35496e801cf46c19dd03d5","id":"efd16c8eac35496e801cf46c19dd03d5","name":"default"},"profile":{"href":"https://us-south.iaas.cloud.ibm.com/v1/volume/profiles/general-purpose","name":"general-purpose"},"created_at":"2019-04-05T05:32:02Z","status":"pending","volume_attachments":[],"zone":{"name":"us-south-1","href":"https://us-south.iaas.cloud.ibm.com/v1/regions/us-south/zones/us-south-1"},"crn":"crn:v1:bluemix:public:is:us-south-1:a/e242f140687cd68a8e037b26680e0f04::volume:0f943442-4c42-4e07-beca-cde25c64e5f5"}}
{"level":"info","timestamp":"2019-04-05T11:02:34.547+0530","caller":"provider/get_volume.go:51","msg":"Exit GetVolume","name":"ibmcloud-storage-volume-lib/main","VolumeLib":"IBMCloud-Storage-Lib"}
2019/04/05 11:02:36 DeleteVolume took 1.905956331s
DeleteVolume took 4.028752705s
{"level":"info","timestamp":"2019-04-05T11:03:10.485+0530","caller":"provider/delete_volume.go:38","msg":"Exit DeleteVolume()","name":"ibmcloud-storage-volume-lib/main","VolumeLib":"IBMCloud-Storage-Lib"}
```

**Comment by narkarum** [comment]
```
{"level":"info","timestamp":"2019-04-05T11:36:37.550+0530","caller":"provider/create_volume.go:23","msg":"Entry CreateVolume","name":"ibmcloud-storage-volume-lib/main","VolumeLib":"IBMCloud-Storage-Lib","volumeRequest":{"provider":"","volumeType":"vpc-block","capacity":10,"snapshotSpace":10,"iops":"0","tier":null,"az":"us-south-1","creationTime":"0001-01-01T00:00:00Z","serviceOffering":null,"name":"simha13","resource_group":{"id":"efd16c8eac35496e801cf46c19dd03d5"},"generation":"gt","profile":{"name":"general-purpose"},"tags":["Testing VPC Volume"]}}
{"level":"info","timestamp":"2019-04-05T11:36:37.550+0530","caller":"provider/create_volume.go:26","msg":"Creating volume as per order request .... ","name":"ibmcloud-storage-volume-lib/main","VolumeLib":"IBMCloud-Storage-Lib","volumeRequest":{"provider":"","volumeType":"vpc-block","capacity":10,"snapshotSpace":10,"iops":"0","tier":null,"az":"us-south-1","creationTime":"0001-01-01T00:00:00Z","serviceOffering":null,"name":"simha13","resource_group":{"id":"efd16c8eac35496e801cf46c19dd03d5"},"generation":"gt","profile":{"name":"general-purpose"},"tags":["Testing VPC Volume"]}}
{"level":"info","timestamp":"2019-04-05T11:36:37.550+0530","caller":"provider/create_volume.go:76","msg":"Volume request details","name":"ibmcloud-storage-volume-lib/main","VolumeLib":"IBMCloud-Storage-Lib","Volume request template":{"name":"simha13","capacity":10,"resource_group":{"id":"efd16c8eac35496e801cf46c19dd03d5"},"tags":["Testing VPC Volume"],"generation":"gt","profile":{"name":"general-purpose"},"zone":{"name":"us-south-1"}}}
2019/04/05 11:36:40 CreateVolume took 2.98733925s
{"level":"info","timestamp":"2019-04-05T11:36:40.537+0530","caller":"provider/create_volume.go:86","msg":"Created volume details","name":"ibmcloud-storage-volume-lib/main","VolumeLib":"IBMCloud-Storage-Lib","Volume":{"href":"https://us-south.iaas.cloud.ibm.com/v1/volumes/fc721050-2082-44a3-975e-6ec5926fb35b","id":"fc721050-2082-44a3-975e-6ec5926fb35b","name":"simha13","capacity":10,"iops":3000,"resource_group":{"href":"https://resource-manager.bluemix.net/v1/resource_groups/efd16c8eac35496e801cf46c19dd03d5","id":"efd16c8eac35496e801cf46c19dd03d5","name":"default"},"profile":{"href":"https://us-south.iaas.cloud.ibm.com/v1/volume/profiles/general-purpose","name":"general-purpose"},"created_at":"2019-04-05T06:06:39Z","status":"pending","volume_attachments":[],"zone":{"name":"us-south-1","href":"https://us-south.iaas.cloud.ibm.com/v1/regions/us-south/zones/us-south-1"},"crn":"crn:v1:bluemix:public:is:us-south-1:a/e242f140687cd68a8e037b26680e0f04::volume:fc721050-2082-44a3-975e-6ec5926fb35b"}}
{"level":"info","timestamp":"2019-04-05T11:36:40.537+0530","caller":"provider/get_volume.go:23","msg":"Entry GetVolume","name":"ibmcloud-storage-volume-lib/main","VolumeLib":"IBMCloud-Storage-Lib","VolumeID":"fc721050-2082-44a3-975e-6ec5926fb35b"}
2019/04/05 11:36:42 GetVolume took 1.953416784s
{"level":"info","timestamp":"2019-04-05T11:36:42.491+0530","caller":"provider/get_volume.go:38","msg":"Volume details","name":"ibmcloud-storage-volume-lib/main","VolumeLib":"IBMCloud-Storage-Lib","Volume":{"href":"https://us-south.iaas.cloud.ibm.com/v1/volumes/fc721050-2082-44a3-975e-6ec5926fb35b","id":"fc721050-2082-44a3-975e-6ec5926fb35b","name":"simha13","capacity":10,"iops":3000,"resource_group":{"href":"https://resource-manager.bluemix.net/v1/resource_groups/efd16c8eac35496e801cf46c19dd03d5","id":"efd16c8eac35496e801cf46c19dd03d5","name":"default"},"profile":{"href":"https://us-south.iaas.cloud.ibm.com/v1/volume/profiles/general-purpose","name":"general-purpose"},"created_at":"2019-04-05T06:06:39Z","status":"pending","volume_attachments":[],"zone":{"name":"us-south-1","href":"https://us-south.iaas.cloud.ibm.com/v1/regions/us-south/zones/us-south-1"},"crn":"crn:v1:bluemix:public:is:us-south-1:a/e242f140687cd68a8e037b26680e0f04::volume:fc721050-2082-44a3-975e-6ec5926fb35b"}}
{"level":"info","timestamp":"2019-04-05T11:36:42.491+0530","caller":"provider/get_volume.go:51","msg":"Exit GetVolume","name":"ibmcloud-storage-volume-lib/main","VolumeLib":"IBMCloud-Storage-Lib","VolumeID":"fc721050-2082-44a3-975e-6ec5926fb35b"}
{"level":"info","timestamp":"2019-04-05T11:36:42.491+0530","caller":"provider/create_volume.go:91","msg":"Exit CreateVolume","name":"ibmcloud-storage-volume-lib/main","VolumeLib":"IBMCloud-Storage-Lib","volumeRequest":{"provider":"","volumeType":"vpc-block","capacity":10,"snapshotSpace":10,"iops":"0","tier":null,"az":"us-south-1","creationTime":"0001-01-01T00:00:00Z","serviceOffering":null,"name":"simha13","resource_group":{"id":"efd16c8eac35496e801cf46c19dd03d5"},"generation":"gt","profile":{"name":"general-purpose"},"tags":["Testing VPC Volume"]}}
{"level":"info","timestamp":"2019-04-05T11:36:42.491+0530","caller":"ibmcloud-storage-volume-lib/main.go:338","msg":"Successfully ordered volume ================>","name":"ibmcloud-storage-volume-lib/main","VolumeLib":"IBMCloud-Storage-Lib","volumeObj":{"volumeID":"fc721050-2082-44a3-975e-6ec5926fb35b","provider":"VPC","volumeType":"vpc-block","capacity":10,"iops":"3000","tier":null,"region":"us-south-1","creationTime":"2019-04-05T06:06:39Z","serviceOffering":null}}
```

File: volume-providers/vpc/vpcclient/vpcvolume/util.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	"fmt"
+	"log"
+	"time"
+)
+
+// Get execution time of a function
+func TimeTrack(functionName string, start time.Time) {
+	elapsed := time.Since(start)
+
+	log.Println(fmt.Sprintf("%s took %s", functionName, elapsed))
```
[the remark this answers is not in the document]
Comment (narkarum) [reply volume-providers/vpc/vpcclient/vpcvolume/util.go]: This is a client code, logger is not added.

**Pull request merged**